### PR TITLE
Quote vagrant floc 2871

### PR DESCRIPTION
**Pull request by adamtheturtle**
Fixes https://clusterhq.atlassian.net/browse/FLOC-2871

This is how it fails on master on the regular Buildbot:

```
~$ vagrant box add http://build.clusterhq.com/results/vagrant/master/flocker-tutorial.json
==> box: Loading metadata for box 'http://build.clusterhq.com/results/vagrant/master/flocker-tutorial.json'
==> box: Adding box 'clusterhq/flocker-tutorial' (v1.4.0.331.gba3dcc8) for provider: virtualbox
    box: Downloading: https://s3.amazonaws.com/clusterhq-dev-archive/vagrant/tutorial/flocker-tutorial-1.4.0+331.gba3dcc8.box
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

The requested URL returned error: 403 Forbidden
~$
```

A builder is set up (for now) with this new code at 52.89.42.46

```
~$ vagrant box add http://52.89.42.46/results/vagrant/master/flocker-tutorial.json
==> box: Loading metadata for box 'http://52.89.42.46/results/vagrant/master/flocker-tutorial.json'
==> box: Adding box 'clusterhq/flocker-tutorial' (v1.4.0.331.gba3dcc8) for provider: virtualbox
    box: Downloading: https://s3.amazonaws.com/clusterhq-dev-archive/vagrant/tutorial/flocker-tutorial-1.4.0%2B331.gba3dcc8.box
==> box: Successfully added box 'clusterhq/flocker-tutorial' (v1.4.0.331.gba3dcc8) for 'virtualbox'!
~$
```

[Review on Reviewable](https://reviewable.io/reviews/clusterhq/build.clusterhq.com/143)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -272,16 +272,17 @@ To create a Rackspace OnMetal slave to serve this purpose:
...
-To configure any Fedora 20 bare metal machine (e.g. on OnMetal as above)::
+To configure any Fedora 20 or Fedora 21 bare metal machine (e.g. on OnMetal as above)::
 
+   MASTER=<The Buildbot master's hostname or IP address>
+   ONMETAL_IP_ADDRESS=<The IP address of the OnMetal Server>
+   PASSWORD=<The password in "slaves.fedora-20/vagrant.passwords" from the "config.yml" or "staging.yml" file used to deploy the BuildBot master>
    fab -f slave/vagrant/fabfile.py --hosts=root@${ONMETAL_IP_ADDRESS} install:0,${PASSWORD},${MASTER}
 
-Where ``${PASSWORD}`` is the password in ``slaves.fedora-20/vagrant.passwords`` from the ``config.yml`` or ``staging.yml`` file used to deploy the BuildBot master on hostname or IP address ``${MASTER}``.
-
```
Comment (jml) [inline README.rst]: This seems unrelated to this change. Could you please split it into a separate CL?

File: flocker_bb/builders/flocker_vagrant.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+from urllib import quote
+
 from buildbot.steps.shell import ShellCommand, SetPropertyFromCommand
 from buildbot.process.properties import Interpolate, Property, renderer
```
Comment (jml) [inline flocker_bb/builders/flocker_vagrant.py]: I don't understand why this line is yellow.

File: flocker_bb/builders/flocker_vagrant.py [hunk 3]
```diff
@@ -34,6 +36,14 @@ def render(props):
     return render
 
 
+def quoted_version(version):
+    @renderer
+    def render(props):
+        return (props.render(version)
+                .addCallback(lambda v: quote(v)))
```
Comment (jml) [inline flocker_bb/builders/flocker_vagrant.py]: The `lambda` is superfluous. Just add `quote` as a callback.

File: flocker_bb/builders/flocker_vagrant.py (hunk 2 above)
Comment (adamtheturtle) [inline flocker_bb/builders/flocker_vagrant.py]: Have you clicked the line number?

Comment (jml) [inline flocker_bb/builders/flocker_vagrant.py]: Ah, that'll be it. :grin:

**Comment by jml** [comment]
:+1: if you make the suggested changes. If not, then please resubmit for review.
